modules/lib/src/main/resources/assets/js/app/browse/action/MarkAsReadyContentAction.ts

```diff
@@ -7,48 +7,62 @@ import {showFeedback} from '@enonic/lib-admin-ui/notify/MessageBus';
         this.setEnabled(false);
+        this.canPublish = false;
 
         this.confirmDialog = new ConfirmationDialog().setQuestion(i18n('dialog.markAsReady.question'));
     }
 
     protected handleExecuted() {
-        const contentToMarkAsReady: ContentSummaryAndCompareStatus[] = this.grid.getSelectedDataList().filter(
-            (item: ContentSummaryAndCompareStatus) => item.canBeMarkedAsReady()
-        );
+        const content: ContentSummaryAndCompareStatus[] = this.grid.getSelectedDataList();
+        const contentToMarkAsReady: ContentSummaryAndCompareStatus[] = this.grid.getSelectedDataList()
+            .filter((item: ContentSummaryAndCompareStatus) => item.canBeMarkedAsReady());
         const isSingleItem: boolean = contentToMarkAsReady.length === 1;
 
         if (isSingleItem) {
-            MarkAsReadyContentAction.markAsReady(contentToMarkAsReady);
+            this.markAsReadyAndPublish(contentToMarkAsReady, content);
         } else {
-            this.confirmDialog.setYesCallback(() => {
-                MarkAsReadyContentAction.markAsReady(contentToMarkAsReady);
-            }).open();
+            this.confirmDialog.setYesCallback(() => void this.markAsReadyAndPublish(contentToMarkAsReady, content)).open();
         }
     }
 
-    private static markAsReady(content: ContentSummaryAndCompareStatus[]): Q.Promise<void> {
-        const contentIds = content.map(item => item.getContentId());
-        const isSingleItem = content.length === 1;
+    private markAsReadyAndPublish(
+        contentToMark: ContentSummaryAndCompareStatus[],
+        contentToPublish: ContentSummaryAndCompareStatus[]
+    ): Q.Promise<void> {
+        const contentIds: ContentId[] = contentToMark.map(item => item.getContentId());
+        const isSingleItem: boolean = contentToMark.length === 1;
         return new MarkAsReadyRequest(contentIds).sendAndParse().then(() => {
             if (isSingleItem) {
-                const name = content[0].getContentSummary().getName();
+                const name = contentToMark[0].getContentSummary().getName();
                 showFeedback(i18n('notify.item.markedAsReady', name));
             } else {
-                showFeedback(i18n('notify.item.markedAsReady.multiple', content.length));
+                showFeedback(i18n('notify.item.markedAsReady.multiple', contentToMark.length));
+            }
+        }).then(() => {
+            if (this.canPublish) {
+                new ContentPublishPromptEvent({model: contentToPublish}).fire();
             }
         }).catch(DefaultErrorHandler.handle);
     }
 
     isToBeEnabled(state: ContentTreeGridItemsState): boolean {
+        this.canPublish = !state.isEmpty() && !state.isManagedActionExecuting() && state.canPublish() &&
+                          ((!state.hasAllOnline() && state.hasValidNonOnline()) || state.hasAllPendingDelete()) &&
+                          (state.canModify() || !state.hasAnyInProgress());
+
```

Comment:
This works 100% as intended, but we need to refactor a little. As the condition you assign here to `this.canPublish` is taken from `PublishContentAction`, it makes sense to move it to `ContentTreeGridItemsState`:
```
    isReadyForPublishing(): boolean {
        return !this.isEmpty() && !this.isManagedActionExecuting() && this.canPublish() &&
                !this.hasAllOnline() && this.hasValidNonOnline() &&
                (this.canModify() || !this.hasAnyInProgress());
    }
```

Note that `this.hasAllPendingDelete()` is removed from the check as we no longer support pending deletion.

Then we can use it in both `MarkAsReadyContentAction` and `PublishContentAction`:

`PublishContentAction`:
```
    isToBeEnabled(state: ContentTreeGridItemsState): boolean {
        return state.isReadyForPublishing();
    }
```
`MarkAsReadyContentAction`:

```
    isToBeEnabled(state: ContentTreeGridItemsState): boolean {
        this.canPublish = state.isReadyForPublishing();

        return this.canPublish && state.hasAllValid() && state.canModify() && state.hasAnyCanBeMarkedAsReady();
    }

```